jerry-core/ecma/operations/ecma-conversion.c

```diff
@@ -680,193 +679,184 @@ ecma_op_to_property_descriptor (ecma_value_t obj_value, /**< object value */
                                                                                   if return value is normal
                                                                                   empty completion value */
 {
-  ecma_value_t ret_value = ECMA_VALUE_EMPTY;
-
   /* 1. */
   if (!ecma_is_value_object (obj_value))
   {
-    ret_value = ecma_raise_type_error (ECMA_ERR_MSG ("Expected an object."));
+    return ecma_raise_type_error (ECMA_ERR_MSG ("Expected an object."));
   }
-  else
-  {
-    ecma_object_t *obj_p = ecma_get_object_from_value (obj_value);
 
-    /* 2. */
-    ecma_property_descriptor_t prop_desc = ecma_make_empty_property_descriptor ();
+  ecma_object_t *obj_p = ecma_get_object_from_value (obj_value);
+  ecma_value_t ret_value = ECMA_VALUE_ERROR;
 
-    /* 3. */
-    ECMA_TRY_CATCH (enumerable_prop_value,
-                    ecma_op_object_find (obj_p, ecma_get_magic_string (LIT_MAGIC_STRING_ENUMERABLE)),
-                    ret_value);
+  /* 3. */
+  ecma_value_t enumerable_prop_value = ecma_op_object_find (obj_p,
+                                                            ecma_get_magic_string (LIT_MAGIC_STRING_ENUMERABLE));
 
-    if (ecma_is_value_found (enumerable_prop_value))
-    {
-      uint32_t is_enumerable = (ecma_op_to_boolean (enumerable_prop_value) ? ECMA_PROP_IS_ENUMERABLE
-                                                                           : ECMA_PROP_NO_OPTS);
+  if (ECMA_IS_VALUE_ERROR (enumerable_prop_value))
+  {
+    return enumerable_prop_value;
+  }
 
-      prop_desc.flags |= (uint16_t) (ECMA_PROP_IS_ENUMERABLE_DEFINED | is_enumerable);
-    }
+  /* 2. */
+  ecma_property_descriptor_t prop_desc = ecma_make_empty_property_descriptor ();
 
-    ECMA_FINALIZE (enumerable_prop_value);
+  if (ecma_is_value_found (enumerable_prop_value))
+  {
+    uint32_t is_enumerable = (ecma_op_to_boolean (enumerable_prop_value) ? ECMA_PROP_IS_ENUMERABLE
+                                                                         : ECMA_PROP_NO_OPTS);
 
-    if (!ECMA_IS_VALUE_ERROR (ret_value))
-    {
-      JERRY_ASSERT (ecma_is_value_empty (ret_value));
+    prop_desc.flags |= (uint16_t) (ECMA_PROP_IS_ENUMERABLE_DEFINED | is_enumerable);
+  }
 
-      /* 4. */
-      ECMA_TRY_CATCH (configurable_prop_value,
-                      ecma_op_object_find (obj_p, ecma_get_magic_string (LIT_MAGIC_STRING_CONFIGURABLE)),
-                      ret_value);
+  ecma_free_value (enumerable_prop_value);
```

Comment:
These should only be freed inside the `if (ecma_is_value_found (enumerable_prop_value))`. Same for all below.